dask_xgboost/core.py

```diff
@@ -103,23 +103,39 @@ def train_part(
     args = [("%s=%s" % item).encode() for item in env.items()]
     xgb.rabit.init(args)
     try:
+        local_history = {}
         logger.info("Starting Rabit, Rank %d", xgb.rabit.get_rank())
-
-        bst = xgb.train(param, dtrain, evals=evals, **kwargs)
+        bst = xgb.train(
+            param, dtrain, evals=evals, evals_result=local_history, **kwargs
+        )
 
         if xgb.rabit.get_rank() == 0:  # Only return from one worker
             result = bst
+            evals_result = local_history
         else:
             result = None
+            evals_result = None
     finally:
         xgb.rabit.finalize()
-    return result
+    return result, evals_result
 
 
 def _package_evals(
     eval_set, sample_weight_eval_set=None, missing=None, n_jobs=None
 ):
     if eval_set is not None:
+        eval_set_dask_collections = [
```

Comment:
Why is this check needed now?

I assume that a small, in-memory eval set is the most common scenario, so this is probably fine.

Reply:
If we keep this, I think it can be changed to something like

```python
if any(dask.is_dask_collection(e) for e in evals for evals in evals_set):`
```

Reply:
And can you add a test ensuring that an error is raised?

Reply:
I don't have any strong feelings on where this should be. I had initially stuck it in the XGBClassifier and XGBRegressor `fit` methods, but didn't like duplicating code. 

didn't realize `dask.is_dask_collection(e)` was a thing, thats awesome! I'll update the check and add a unit test

Reply:
I moved the eval_set check to the _train method. This way it'll only get run once and not once per worker. I think this makes a bit more sense.